GTHC/web
# Adding Availability components to client and server side

## Description
This adds the Availability components to the frontend and adds column to each user in the database that holds this information. Also, this PR will allow users to make changes to their availability data from the client side to the server side. This does **NOT** involve the algorithm that is involved to calculate who is available when creating shifts.

## Motivation and Context
This is the the foundation of the full functionality of the availability feature. It is necessary that the availability data can be accessed client side from the server through GET, PUT, and POST requests.

## How Has This Been Tested?
1. Create user with sign up prompts, and see and fill out the new addition availability step
2. Once signed up, go to User Profile and click "Edit User Availability".
3. See how availability is the same as the information you inputted when signed up.
4. Change availability and then click "Save".
5. Refresh page and see updates reflecting on new availability data.





## Screenshots (if appropriate):

![image](https://user-images.githubusercontent.com/13587407/50404561-67790d80-0777-11e9-8485-188a9a253b4b.png)


![image](https://user-images.githubusercontent.com/13587407/50404542-38fb3280-0777-11e9-81be-0a2e7f018e0b.png)


## Types of changes


## Checklist:
